### PR TITLE
MINIFI-270: Move Base RPG capabilities back into RemoteProcessGroupPort

### PR DESCRIPTION
**Pull request by phrocker**
This allows us to extend capabilities in RPG into other classes. Since
the same basic functions are the same we can bring the code up from
Processor into RemoteProcessGroupPort.h instead of a separate class.

Updated LICENSE and added concurrentqueue to locking for the object pool.

Thank you for submitting a contribution to Apache NiFi - MiNiFi C++.

In order to streamline the review of the contribution we ask you
to ensure the following steps have been taken:

### For all changes:
- [ ] Is there a JIRA ticket associated with this PR? Is it referenced
     in the commit message?

- [ ] Does your PR title start with MINIFI-XXXX where XXXX is the JIRA number you are trying to resolve? Pay particular attention to the hyphen "-" character.

- [ ] Has your PR been rebased against the latest commit within the target branch (typically master)?

- [ ] Is your initial contribution a single, squashed commit?

### For code changes:
- [ ] If adding new dependencies to the code, are these dependencies licensed in a way that is compatible for inclusion under [ASF 2.0](http://www.apache.org/legal/resolved.html#category-a)?
- [ ] If applicable, have you updated the LICENSE file?
- [ ] If applicable, have you updated the NOTICE file?

### For documentation related changes:
- [ ] Have you ensured that format looks appropriate for the output in which it is rendered?

### Note:
Please ensure that once the PR is submitted, you check travis-ci for build issues and submit an update to your PR as soon as possible.

### CODE REVIEW

File: LICENSE [hunk 1]
```diff
@@ -401,6 +401,33 @@ ADVISED OF THE POSSIBILITY OF SUCH DAMAGE.
 The views and conclusions contained in the software and documentation are those of the authors and
 should not be interpreted as representing official policies, either expressed or implied, of Dmitry Vyukov.
 
+
+This product bundles 'concurrentqueue' which is available a Boost Software License.
```
Comment (apiri) [inline LICENSE]: The included header file indicates it is a Simplified/2-Clause BSD.  Is there something else that suggests this is BSL?

File: libminifi/src/core/Processor.cpp [hunk 2]
```diff
@@ -194,52 +193,10 @@ void Processor::removeConnection(std::shared_ptr<Connectable> conn) {
   }
 }
 
-std::shared_ptr<Site2SiteClientProtocol> Processor::obtainSite2SiteProtocol(
-    const std::shared_ptr<io::StreamFactory> &stream_factory, std::string host, uint16_t sport, uuid_t portId) {
-  std::lock_guard < std::mutex > lock(mutex_);
+/*
```
Comment (apiri) [inline libminifi/src/core/Processor.cpp]: Remnants of commented out block

File: LICENSE (hunk 1 above)
Comment (apiri) [inline LICENSE]: After actually visiting the linked article on the emotional dromedary, I see that it is actually dual licensed Boost and 2-Clause BSD.

File: libminifi/src/core/Processor.cpp (hunk 2 above)
Comment (phrocker) [inline libminifi/src/core/Processor.cpp]: Sorry I was still self reviewing this.

File: LICENSE (hunk 1 above)
Comment (phrocker) [inline LICENSE]: https://github.com/cameron314/concurrentqueue/blob/master/LICENSE.md

File: libminifi/src/core/Processor.cpp (hunk 2 above)
Comment (phrocker) [inline libminifi/src/core/Processor.cpp]: but thanks!

File: LICENSE (hunk 1 above)
Comment (phrocker) [inline LICENSE]: I'll add a link to that on the wiki entry from other PR.

**Pull request closed without merging**